File: backend/routes/llm.py
from flask import Blueprint, request, jsonify
from transformers import pipeline

llm_blueprint = Blueprint('llm', __name__)

# Load pre-trained models
translator = pipeline('translation_en_to_sw', 'Helsinki-NLP/opus-mt-en-sw')
chatbot = pipeline('conversational', 'microsoft/DialoGPT-medium')

@llm_blueprint.route('/translate', methods=['POST'])
def translate():
    data = request.json
    text = data.get('text')
    try:
        result = translator(text)
        return jsonify({"translatedText": result[0]['translation_text']})
    except Exception as e:
        return jsonify({"error": str(e)}), 500

@llm_blueprint.route('/chat', methods=['POST'])
def chat():
    data = request.json
    message = data.get('message')
    try:
        result = chatbot(message)
        return jsonify({"reply": result[0]['generated_text']})
    except Exception as e:
        return jsonify({"error": str(e)}), 500

File: backend/routes/llm.js
const express = require('express');
const { pipeline } = require('@huggingface/transformers');
const router = express.Router();

// Load a pre-trained model
const translator = pipeline('translation_en_to_sw', 'Helsinki-NLP/opus-mt-en-sw');
const chatbot = pipeline('conversational', 'microsoft/DialoGPT-medium');

router.post('/translate', async (req, res) => {
  const { text } = req.body;
  try {
    const result = await translator(text);
    res.json({ translatedText: result[0].translation_text });
  } catch (error) {
    res.status(500).json({ error: 'Translation failed' });
  }
});

router.post('/chat', async (req, res) => {
  const { message } = req.body;
  try {
    const result = await chatbot(message);
    res.json({ reply: result[0].generated_text });
  } catch (error) {
    res.status(500).json({ error: 'Chat failed' });
  }
});

module.exports = router;
